contrail_api_cli_extra/fix/fix_zk_ip.py

```diff
@@ -103,31 +110,37 @@ def _get_ip_subnet(self, subnets, ip):
     def get_zk_ip(self, vn):
         zk_ips = {}
         subnets = self._get_vn_subnets(vn)
+        zk_subnets_err = []
 
         for s in subnets:
             zk_subnet_req = self._zk_node_for_subnet(vn.fq_name, s)
-            for ip in self.zk_client.get_children(zk_subnet_req):
-                try:
-                    zk_ip = IPAddress(int(ip))
-                except AddrFormatError:
-                    msg = ('{0} INVALID IP : zk_path={1}/{2}'
-                           .format(text_type(ip), zk_subnet_req, text_type(ip)))
-                    logger.warning(msg)
-                    printo(msg)
-                    continue
-                try:
-                    self._get_ip_subnet([s], zk_ip)
-                except SubnetNotFound:
-                    msg = ('{0} OUT OF RANGE {1} : zk_path={2}/{3}'
-                           .format(text_type(zk_ip),
-                                   text_type(s.cidr),
-                                   zk_subnet_req,
-                                   int(zk_ip)))
-                    logger.warning(msg)
-                    printo(msg)
-                    continue
-                zk_ips[zk_ip] = zk_subnet_req + '/' + text_type(int(zk_ip)).zfill(10)
```

Comment:
I would prefer something like:

```
  try:
    children = self.zk_client.get_children(zk_subnet_req)
  except:
    raise ZkNodeNotFound(err)
  for ip in children:
    ....
```
 To avoid nested try/catchs

Reply:
I postpone the raise of the exception to get all the subnet which are not in the ZK server.
Otherwise I can't print the error message inside the **get_zk_ip** function to get each subnet.

Reply:
Then something like:

```
  try:
    children = self.zk_client.get_children(zk_subnet_req)
  except:
    zk_subnets_err.append(zk_subnet_req)
    continue
  for ip in children:
    [....]
  if zk_subnets_err:
￼    raise ZkNodeNotFound(zk_subnets_err) 
```